packages/server/utils/analytics/analytics.ts

```diff
@@ -0,0 +1,101 @@
+import Meeting from '../../database/types/Meeting'
+import MeetingMember from '../../database/types/MeetingMember'
+import MeetingTemplate from '../../database/types/MeetingTemplate'
+import {TeamPromptResponse} from '../../postgres/queries/getTeamPromptResponsesByIds'
+import segment from '../segmentIo'
+import {createMeetingTemplateAnalyticsParams} from './helpers'
+import {SegmentAnalytics} from './segment/SegmentAnalytics'
+
+export type AnalyticsEvent = 'Meeting Started' | 'Meeting Joined' | 'Meeting Completed'
+
+/**
+ * Provides a unified inteface for sending all the analytics events
+ */
+class Analytics {
+  private segmentAnalytics: SegmentAnalytics
+
+  constructor() {
+    this.segmentAnalytics = new SegmentAnalytics(segment)
+  }
+
+  teamPromptEnd = (
+    completedMeeting: Meeting,
+    meetingMembers: MeetingMember[],
+    responses: TeamPromptResponse[]
+  ) => {
+    const userIdsResponses: Record<string, string> = responses.reduce(
+      (previous, response) => ({...previous, [response.userId]: response.plaintextContent}),
+      {}
+    )
+    meetingMembers.forEach((meetingMember) => {
+      const plaintextResponseContent = userIdsResponses[meetingMember.userId]
+      this.meetingEnd(meetingMember.userId, completedMeeting, meetingMembers, {
+        responseAdded: !!plaintextResponseContent
+      })
+    })
+  }
+
+  checkInEnd = (completedMeeting: Meeting, meetingMembers: MeetingMember[]) => {
+    meetingMembers.forEach((meetingMember) =>
+      this.meetingEnd(meetingMember.userId, completedMeeting, meetingMembers)
+    )
+  }
+
+  retrospectiveEnd = (
+    completedMeeting: Meeting,
+    meetingMembers: MeetingMember[],
+    template: MeetingTemplate
+  ) => {
+    meetingMembers.forEach((meetingMember) =>
+      this.meetingEnd(
+        meetingMember.userId,
+        completedMeeting,
+        meetingMembers,
+        createMeetingTemplateAnalyticsParams(template)
+      )
+    )
+  }
+
+  sprintPokerEnd = (
+    completedMeeting: Meeting,
+    meetingMembers: MeetingMember[],
+    template: MeetingTemplate
+  ) => {
+    meetingMembers.forEach((meetingMember) =>
+      this.meetingEnd(
+        meetingMember.userId,
+        completedMeeting,
+        meetingMembers,
+        createMeetingTemplateAnalyticsParams(template)
+      )
+    )
+  }
+
+  private meetingEnd = (
+    userId: string,
+    completedMeeting: Meeting,
+    meetingMembers: MeetingMember[],
+    meetingSpecificProperties?: any
```

Comment:
+1 Could be `Record<string,string | number>`, we probably don't want to do any nesting here.

Reply:
we also send booleans and enums here so I figured it's ok to keep it that way. in different cases we also send arrays, dates etc